Neos.Flow/Classes/Mvc/Routing/Route.php

```diff
@@ -534,10 +534,9 @@ public function resolves(array $routeValues)
                 $routeValues = $internalArguments;
             }
             $queryString = http_build_query($routeValues, null, '&');
-            if ($queryString !== '') {
-                $resolvedUriPath .= strpos($resolvedUriPath, '?') !== false ? '&' . $queryString : '?' . $queryString;
-            }
         }
+
+        $this->resolvedUriConstraints = $this->resolvedUriConstraints->withQueryString($queryString);
```

Comment:
Die Zeile (und damit das klonen der Constraints) wir ja nur benötigt, wenn `$queryString` nicht leer ist.
Darum würde ich vorschlagen, den Check hier drin zu lassen. Oder (vielleicht besser):
`UriConstraints::withQueryString()` so anzupassen, dass es die aktuelle Instanz zurück gibt, wenn der QueryString identisch mit dem aktuellen Wert ist.

Reply:
Right, yes, I can leave that in, I figured it'S rather unnecessary because it defaults to an empty string in 525 so it would just be set to empty string then and teh overheat to do withQueryString seemed not too big. But the condition is no big deal. if you are ok with the addition of the extra condition

Reply:
My point was: Even if it's an empty string, the `UriConstraints` object will be cloned now.
But, sure, that's just a nitpick. The constraint in general makes a lot of sense to me (especially since `path` does _not_ include the query per (PSR) definition)